internal/parser/scanner/scanner_test.go

```diff
@@ -1,15 +1,79 @@
 package scanner
 
 import (
+	"fmt"
+	"math/rand"
 	"testing"
+	"time"
 )
 
 func Test_hasNext(t *testing.T) {
-	t.SkipNow()
+	for _, k := range keywordSlice {
+		input := k
+		scanner := New([]rune(input))
+		for scanner.HasNext() {
+			scanner.Next()
+		}
+	}
+}
 
-	input := "SELECT "
+func Test_specificTokenSequence(t *testing.T) {
+	input := "SelEct FroM"
+	var desiredOutput = []string{"SelEct", "FroM"}
+	var scannerOutput []string
 	scanner := New([]rune(input))
 	for scanner.HasNext() {
-		scanner.Next()
+		token := scanner.Next()
+		if token != nil {
+			scannerOutput = append(scannerOutput, token.Value())
+		}
+	}
+	for i, k := range scannerOutput {
+		if k != desiredOutput[i] {
+			t.Fail()
+		}
+	}
+
+	if t.Failed() {
+		fmt.Printf("Expected %v, obtained %v\n", desiredOutput, scannerOutput)
+	} else {
+		fmt.Printf("Tested on: %v\n", desiredOutput)
+	}
+}
+
+func Test_randomTokenSequence(t *testing.T) {
+	size := 5
+	randomSequence, randomSequenceArray := generateRandomKeywords(size)
```

Comment:
I like the idea, however, this does not generate reproducible tests. Maybe an approach via fuzzing would be better here. But for now this is good.